wait/host.go

```diff
@@ -254,6 +254,26 @@ func UntilUserSignupHasStateLabel(expLabelValue string) UserSignupWaitCriterion
 	}
 }
 
+// WaitForUserSignupsBeingDeleted waits for all UserSignup deletions to be completed
+func (a *HostAwaitility) WaitForUserSignupsBeingDeleted(initialDelay time.Duration) error {
+	time.Sleep(initialDelay)
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		usList := &toolchainv1alpha1.UserSignupList{}
+		if err := a.Client.List(context.TODO(), usList, client.InNamespace(a.Namespace)); err != nil {
+			return false, err
+		}
+		for _, us := range usList.Items {
+			if us.DeletionTimestamp != nil {
+				a.T.Logf("a UserSignup is being deleted: %s", us.Name)
+				return false, err
```

Comment:
Just to improve code readability a bit.
```suggestion
				return false, nil
```